doc/user/formats.rst

```diff
@@ -0,0 +1,129 @@
+Model file formats
+==================
+
+Models are stored in HDF5 files. Where boolean or complex values are stored,
+they use the conventions set by `h5py`_. The extension should be :file:`.h5`
+(:file:`.hdf5` is also accepted by this package). When served over HTTP, the
+``Content-Type`` should be ``application/x-hdf5``. 
+
+.. _h5py: https://docs.h5py.org/en/stable/
+
+The base filename should be a checksum of the file content: this ensures that
+models can be cached indefinitely as a new version of a model is guaranteed to
+have a new checksum. The filename should thus be
+:file:`{algorithm}_{checksum}.h5` where the only currently supported value for
+:samp:`{algorithm}` is ``sha256``. The package verifies the checksum if it is
+recognised, but using a filename without a checksum is not an error.
+
+Alternatively, a model with an extension of :file:`.alias` or a
+``Content-Type``
+of ``text/plain`` is an :dfn:`alias`. It must contain a relative path to the
+actual model. Aliases can also point at other aliases, but there is an
+implementation-defined maximum redirection depth. Aliases may not change
+origin (scheme, host or port); this makes it practical to mirror a set of
+models in a filesystem or on another server without needing to rewrite the
+aliases.
+
+Model files will contain the following attributes on the root (the ``model_*``
+namespace is reserved for the general framework, and shouldn't be used for
+type-specific data):
+
+model_type
+    The model type (see :doc:`concepts`). See later sections for model types
+    defined by this specification. New model types may be defined in future.
+    To avoid collisions, users may define their own types with the prefix
+    ``x_``.
+
+model_format
+    Indicates how the model is encoded in the file, and is specific to each
+    model type. This value also serves as a major version number: if
+    backwards-incompatible changes are made, a new ``model_format`` must be used.
```

Comment:
Could you provide an example of this? Do you mean e.g. a slightly tweaked spline format?

Reply:
Sure, that could be an example.